docs/user/build-customization.rst

```diff
@@ -0,0 +1,155 @@
+Build customization
+===================
+
+Read the Docs has a :ref:`well-defined build process <builds>` that may not be enough for all the use cases.
+This page explains how to customize Read the Docs build process to get the most of it and support custom build processes.
+
+In the normal build process the jobs ``checkout``, ``system_dependencies``, ``create_environment``, ``install``, ``build`` and ``upload`` are executed.
+However, Read the Docs exposes in-between jobs to users so they can customize the build process by running shell commands.
+These in-between jobs are:
+
+:post_checkout:
+
+:pre_system_dependencies:
+
+:post_system_dependencies:
+
+:pre_create_environment:
+
+:post_create_environment:
+
+:pre_install:
+
+:post_install:
+
+:pre_build:
+
+:post_build:
```

Comment:
This renders pretty oddly. I think we just want a bulleted list?

Reply:
Yeah, I'm happy to improve the rendering of this. I was about to add content on each job but then realized that it would be too repetitive and useless. I'll try Anthony's suggestion with the table.